src/rgw/rgw_json_enc.cc

```diff
@@ -2017,6 +2028,17 @@ void LCRule::dump(Formatter *f) const
   f->dump_object("noncur_expiration", noncur_expiration);
   f->dump_object("mp_expiration", mp_expiration);
   f->dump_object("filter", filter);
+  f->open_object_section("transitions");  
+  for(auto iter = transitions.begin(); iter != transitions.end(); iter++){
+    f->dump_object(iter->first.c_str(), iter->second);
+  }
```

Comment:
```suggestion
  for(auto& [storage_class, transition] : transitions){
    f->dump_object(storage_class.c_str(), transition);
  }
```

Reply:
ok, I have added prefix "rgw:", and replaced for(auto iter ...; iter != ...; iter++) {} by for(auto& [.., ..] : ..)